### ISSUE TITLE
[BUG] Transform::default() defaults to f64

### ISSUE THREAD

**Issue by bamling** [amethyst/amethyst#1574]
## Description

I figured I'd follow the pong tutorial again after the `Transform` refactor to see what needs to be updated in the documentation and I came across an issue with `Transform::default()`. `Transform::default()` _defaults_ to `f64` which prevents the early examples from running:

```
thread 'main' panicked at 'Tried to fetch a resource of type "amethyst::specs::storage::MaskedStorage<amethyst::amethyst_core::Transform<f64>>", but the resource does not exist.
```

By the looks of it Amethyst registers a `Transform<f32>` storage by default which clashes with the `Transform::default()` implementation. Using the following work as a workaround, however this should not be required when working with sane defaults:

```
let mut transform: Transform<f32> = Transform::default();
// or
let mut transform = Transform::<f32>::default();
```

## Reproduction Steps

_see above_

## What You Expected to Happen

I'd expect the _defaults_ to **work** across the system without workarounds or explicit type parameters.

## What Actually Happened

_see above_

## Your Environment

Version of Amethyst:
`master` with Commit ID `29b4b8d61c6e2dd9071448bc388a3bf05e29f4ed`

**Comment by jojolepro** [comment]
working on it! :)

**Comment by fhaynes** [comment]
@bamling Thanks for the report. We'll have this fixed before 0.11. Definitely should not have happened.
